pyHS100/smartplug.py

```diff
@@ -80,6 +80,54 @@ def state(self, value: str):
         else:
             raise ValueError("State %s is not valid.", value)
 
+    @property
+    def brightness(self) -> Optional[int]:
+        """
+        Current brightness of the device, if supported
+
+        :returns: integer
+        :rtype: int
+        
+        """
+        if not self.is_dimmable:
+            return None
+
+        return int(self.sys_info['brightness'])
+        
+    @brightness.setter
+    def brightness(self, value: int):
+        """
+        Set the new switch brightness level
+
+        :param value: integer between 1 and 100
+        
+        """
+        if not self.is_dimmable:
+            return None
+
+        if not isinstance(value, int):
+            raise ValueError("Brightness must be integer, not of %s.", type(value))
+        elif value > 0 and value <= 100:
```

Comment:
Just to be sure, the values are between 1-100?

Reply:
Yes, this is correct, I specified valid values to be between 1 and 100.  I did not feel comfortable "assuming" that setting brightness to 0 would turn the light off, versus an explicit call by sending to the "off" command.  I can test this tonight by sending a value of 0 for brightness to see if the "relay_state" automatically switches of 0 when / if brightness is set to 0 before, and if so, allow 0 before this is merged.

Reply:
Okay, I VPN'd home from work and can indeed confirm that allowing a brightness of 0 will NOT turn the switch off and in addition will NOT change whatever the current brightness level is (i.e. if brightness is set to 60, and a value of 0 is passed in, it will stay at 60).  Furthermore, the way things are currently coded, by allowing 0 as a brightness value, this would turn the light on to whatever its previous brightness value is set to, since 0 has no impact.  So, I think things are good they way they are.